app/src/main/java/com/example/poste/models/User.java

```diff
@@ -246,7 +247,100 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
     }
 
     /**
-     * Parses the error response from the backend. This function expects the error response in the
+     * Deletes a post from the backend API.
+     *
+     * This method uses the {@link PostDeletionCallback} interface to handle the results of the
+     * asynchronous API call. This interface defines two methods:
+     * {@link PostDeletionCallback#onSuccess()} and {@link PostDeletionCallback#onError(String)}.
+     *
+     * By implementing this interface, you can define the desired actions based on the deletion
+     * outcome by overriding these methods. For example, if this is called in an Activity, you can
+     * display a Toast message in the {@link PostDeletionCallback#onSuccess()} method to notify the
+     * user that the post was successfully deleted, or display a Toast message in the
+     * {@link PostDeletionCallback#onError(String)} method to notify the user that the post could
+     * not be deleted.
+     *
+     * Example of creating and passing a callback, for instance in an Activity:
+     *
+     * <pre>
+     * {@code
+     * PostDeletionCallback callback = new PostDeletionCallback() {
+     *     @Override
+     *     public void onSuccess() {
+     *         // e.g. Display a Toast message for successful deletion. What you want to happen
+     *         // after a successful deletion should be defined here. Note that this method
+     *         // already removes the post from the user's folders, so you don't need to do that
+     *         // here.
+     *     }
+     *
+     *     @Override
+     *     public void onError(String errorMessage) {
+     *         // e.g. Display a Toast message for errors. What you want to happen after an error
+     *         // should be defined here.
+     *     }
+     * };
+     *
+     * User user = User.getUser();
+     * user.deletePost(postToBeDeleted, callback);
+     * }
+     * </pre>
+     *
+     * @param post The Post model to delete.
+     * @param callback The callback to be invoked after attempting to delete the post.
+     */
+    public void deletePostFromServer(Post post, PostDeletionCallback callback) {
+        String id = post.getId();
+        Log.d("UserDebug", "Attempting to delete post ID " + id);
+        if (id == null) {
+            callback.onError("Post ID is null");
+            Log.d("UserDebug", "Error deleting post: post ID is null");
+            return;
+        }
+        
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Log.d("UserDebug", "Sending delete request to API for Post ID " + id);
+        Call<ResponseBody> call = apiService.deletePost(getTokenHeader(), id);
+
+        call.enqueue(new Callback<ResponseBody>() {
```

Comment:
Here, we see we've actually been using Callbacks all along, without much consideration to their meaning